util/dvsim/dvsim.py

```diff
@@ -572,6 +577,13 @@ def main():
         cfg.print_list()
         sys.exit(0)
 
+    # If --cov-unr is passed, run UNR to generate report for unreachable
+    # exclusion file.
+    if args.cov_unr:
+        cfg.cov_unr()
+        cfg.deploy_objects()
+        sys.exit(0)
```

Comment:
I feel like this part of dvsim's command handling is a little problematic. The issue is that we now have a load of code like this:
```python
if has_arg_foo():
    do_something()
    exit(0)

if has_arg_bar():
    do_something()
    exit(0)

if has_arg_baz():
    do_something()
    exit(0)
```
This allows you to run `my_tool --bar --foo --baz`. But what will happen? In this example, the same thing as if I ran `my_tool --foo`, but I can only tell by looking at the code.

I think the "official" way to avoid this problem is with argparse's "[mutually exclusive groups](https://docs.python.org/3/library/argparse.html#mutual-exclusion)". Unfortunately, these don't seem to play well with argument groups otherwise (I don't think you can have a "mutually exclusive subgroup", nor can you have mutually exclusive arguments from two different groups).

So I think we have to do something a little more manual. Could you try adding a check to make sure that `args.cov_analyze`, `args.purge` and `args.items` are not set (since those are the things we'll definitely ignore below)?

If the code turns out really ugly, don't worry: we can leave things as is for now, and try to come up with a nicer solution in the future :-)

Reply:
Good question. I moved `args.purge` part above the `cov_analyze` and `cov_unr` as we can still clean up scratch for these 2 commands.

But most of the options don't work with `cov_analyze` and `cov_unr`, like waveform related option, simulation related (run-opts, verbosity, seed, reseed, xprop etc). Not sure what is the good way to check if the input options are correct. Now we have `cov_analyze` and `cov_unr` on the top, which can silently ignore the other unrelated options.
@sriyerg any thought?

Reply:
Yeah, I'm also not sure of the best way to do this. It can definitely wait for a later PR. I suppose we could do something like define "compatibility" for arguments that allows you to say things like "--foo can only be used with --bar or --baz" or "--qux can be used with anything but --blargl". Then we just have some generic code that checks all the supplied options make sense together.

I wonder whether there are other programs with lots of options, and whether we can steal their UI :-) I guess that many have gone down the subprogram route (so you'd have "dvsim.py cov_unr --some --arguments" and similar). That sorts out the mutually exclusive thing, but it's less generic than what we've got at the moment.

Reply:
Sure, we can figure this out later. Thanks @rswarbrick